packages/language-client/index.js

```diff
@@ -0,0 +1,43 @@
+import {workspace} from 'vscode'
+import {LanguageClient} from 'vscode-languageclient/node.js'
+
+/**
+ * @type {LanguageClient}
+ */
+let client
+
+/**
+ * @param {import('vscode').ExtensionContext} context
+ */
+export async function activate(context) {
+  if (!workspace.getConfiguration('mdx').get('experimentalLanguageServer')) {
+    return
+  }
+
+  const module = context.asAbsolutePath('out/language-server.js')
+
+  client = new LanguageClient(
+    'MDX',
+    {
+      run: {module},
+      debug: {module, options: {execArgv: ['--inspect=6009', '--nolazy']}}
+    },
+    {
+      documentSelector: [
+        {scheme: 'file', language: 'mdx'},
+        {scheme: 'file', language: 'typescript'},
+        {scheme: 'file', language: 'typescriptreact'},
+        {scheme: 'file', language: 'javascript'},
+        {scheme: 'file', language: 'javascriptreact'}
+      ]
+    }
+  )
+
+  await client.start()
+}
+
+export async function deactivate() {
+  if (client) {
+    await client.stop()
```

Comment:
should `client` be set to undefined, so it isn’t accidentally hanging around/stopped twice?

Reply:
Good question! This is pretty much a text book example. Your point makes sense, but I don’t see them set `client` to undefined in https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#explaining-the-language-client.

I also considered the following alternative:

```js
  const client = new LanguageClient(/* … */)

  await client.start()

  context.subscriptions.push(client)
```

Reply:
I’m marking this as resolved, because although I agree it appears a bit odd, I don’t think this is a problem.